crates/ark/src/data_explorer/format.rs

```diff
@@ -709,4 +738,33 @@ mod tests {
             ]);
         })
     }
+
+    #[test]
+    fn test_truncation() {
+        r_test(|| {
+            let mut options = default_options();
+            options.max_value_length = 3;
+
+            let data = r_parse_eval0(r#"c("aaaaa", "aaaaaaaa", "aa")"#, R_ENVS.global).unwrap();
+            let formatted = format_column(data.sexp, &options);
+            assert_eq!(formatted, vec![
+                ColumnValue::FormattedValue("aaa".to_string()),
+                ColumnValue::FormattedValue("aaa".to_string()),
+                ColumnValue::FormattedValue("aa".to_string()),
```

Comment:
On the frontend, does the user get any indication of the fact that the value was truncated? It might be nice for them to have some visual indication of this, like a trailing `...` or something.

I assume that would mean the backend would have to send over some details in `ColumnValue::FormattedValue` like `truncated: true` to tell the frontend that it was truncated. May be worth it, not sure. I can just imagine some confusion coming from the fact that it could _look_ like you have all the values, but in reality some truncation happened

Reply:
Currently with the default option to `1000` (set by the frot-end) it is actually impossible for a user to see the end of the string, so the truncation is only there as a safety feature to avoid sending a huge amount of data to the front-end in case there's a single cell with a huge text inside.